src/plugins/stylesheet/CSSResourcePlugin.ts

```diff
@@ -190,16 +203,54 @@ export class CSSResourcePluginClass implements Plugin {
             if (this.distFolder) {
                 let newFileName = this.useOriginalFilenames ? path.relative(file.context.homeDir, urlFile) : generateNewFileName(urlFile);
 
-                if (!files[urlFile]) {
+                if (!this.files[urlFile]) {
                     let newPath = path.join(this.distFolder, newFileName);
                     tasks.push(copyFile(urlFile, newPath));
-                    files[urlFile] = true;
+                    this.files[urlFile] = true;
+                    
+                    // We store this copied file source and destination path for the middleware
+                    this.copiedFiles.push({
+                        from: urlFile,
+                        to: newPath
+                    });
+
+                    // We also store a string which uniquely identify this array
+                    // To avoid watch loop
+                    this.copiedFilesID = this.copiedFiles.map(
+                        copiedFile => generateNewFileName( copiedFile.from )
```

Comment:
Why are we generating a new file name if it has been generated above? Besides, user might choose to use originalFileNames

Reply:
In fact we are using generateNewFileName to create a hash of the copiedFiles array. It is needed to identify when the list is updated. Maybe there is a better solution here.

Also, originalFileNames is not enough when we want generated file names and also JS aware of generated resources. This is why this option is needed.

Reply:
How about checking for the option enabled? We don't need to generate those unless the option is set. Right?

Reply:
Right ! I'll create and test a cleaner pull request, without this copiedFileID logic. In fact, CSSResourcesPlugin only add resources to the list (removals are not detected), so only checking array length (and option enabled) should be enough.